backend/internal/user/handler.go

```diff
@@ -387,11 +388,26 @@ func (uh *userHandler) HandleSelfUserCredentialUpdateRequest(w http.ResponseWrit
 		return
 	}
 
-	if svcErr := uh.userService.UpdateUserCredentials(userID, updateRequest.Attributes); svcErr != nil {
-		handleError(w, svcErr)
+	// Parse the attributes to get credential types and values
+	var attrs map[string]json.RawMessage
+	if err := json.Unmarshal(updateRequest.Attributes, &attrs); err != nil {
+		handleError(w, &ErrorInvalidRequestFormat)
 		return
 	}
 
+	if len(attrs) == 0 {
+		handleError(w, &ErrorMissingCredentials)
+		return
+	}
+
+	// Update each credential type (replace mode by default for user-initiated updates)
+	for credentialType, credentialValue := range attrs {
+		if svcErr := uh.userService.UpdateUserCredentials(userID, credentialType, credentialValue, false); svcErr != nil {
+			handleError(w, svcErr)
+			return
+		}
+	}
```

Comment:
I don't think performing a for loop here is not the best approach. Can't we have the interface method to accepts credentials list ?